plugins/jobs/web_client/models/JobModel.js

```diff
@@ -36,8 +36,9 @@ var JobModel = AccessControlledModel.extend({
             }));
         }
 
-        this.set('segments', segments);
```

Comment:
Yes, per my [commit comment](), `'segments'` is not a Model attribute. It doesn't exist on the server and doesn't store any unique information. We should not abuse Model attributes to serve as a cache for transformed data, as it will cause problems with `Model.sync`, etc. (should we ever want to start using it property) and make our models harder to reason about (since the value gets populated on-demand, not as part of the natural Model lifecycle).

The old control flow wasn't really pre-computing this value at a more convenient point in the lifecycle or a fewer number of times; it was just calling this once per job, then later accessing the attribute once per job. So, rather than pointlessly cache the value, I just made this into a utility.